babyagi.py

```diff
@@ -1,59 +1,59 @@
 #!/usr/bin/env python3
-import argparse
 import os
+import time
 import openai
 import pinecone
-import time
-import sys
 from collections import deque
 from typing import Dict, List
 from dotenv import load_dotenv
-import os
-
-# Parse arguments for optional extensions
-parser = argparse.ArgumentParser()
-parser.add_argument('-e', '--env', nargs='+', help='filenames for env')
-args = parser.parse_args()
 
 # Load default environment variables (.env)
 load_dotenv()
 
-# Set environment variables for optional extensions
-if args.env:
-    for env_path in args.env:
-        load_dotenv(env_path)
-        print('Using env from file:', env_path)
-
 # Set API Keys
 OPENAI_API_KEY = os.getenv("OPENAI_API_KEY", "")
 assert OPENAI_API_KEY, "OPENAI_API_KEY environment variable is missing from .env"
 
 OPENAI_API_MODEL = os.getenv("OPENAI_API_MODEL", "gpt-3.5-turbo")
-assert OPENAI_API_MODEL, "OPENAI_API_MODEL environment variable is missing from .env"
-
-if "gpt-4" in OPENAI_API_MODEL.lower():
-    print(f"\033[91m\033[1m"+"\n*****USING GPT-4. POTENTIALLY EXPENSIVE. MONITOR YOUR COSTS*****"+"\033[0m\033[0m")
 
 PINECONE_API_KEY = os.getenv("PINECONE_API_KEY", "")
 assert PINECONE_API_KEY, "PINECONE_API_KEY environment variable is missing from .env"
 
 PINECONE_ENVIRONMENT = os.getenv("PINECONE_ENVIRONMENT", "us-east1-gcp")
-assert PINECONE_ENVIRONMENT, "PINECONE_ENVIRONMENT environment variable is missing from .env"
 
 # Table config
 YOUR_TABLE_NAME = os.getenv("TABLE_NAME", "")
 assert YOUR_TABLE_NAME, "TABLE_NAME environment variable is missing from .env"
 
-# Project config
+# Goal configuation
 OBJECTIVE = os.getenv("OBJECTIVE", "")
-assert OBJECTIVE, "OBJECTIVE environment variable is missing from .env"
+INITIAL_TASK = os.getenv("INITIAL_TASK", os.getenv("FIRST_TASK", ""))
+
+DOTENV_EXTENSIONS = os.getenv("DOTENV_EXTENSIONS", "").split(' ')
+
+# Command line arguments extension
+# Can override any of the above environment variables
+ENABLE_COMMAND_LINE_ARGS = os.getenv("ENABLE_COMMAND_LINE_ARGS", "false").lower() == "true"
+if ENABLE_COMMAND_LINE_ARGS:
+    from extensions.argparseext import parse_arguments
+    OBJECTIVE, INITIAL_TASK, OPENAI_API_MODEL, DOTENV_EXTENSIONS = parse_arguments()
+
+# Load additional environment variables for enabled extensions
+if DOTENV_EXTENSIONS:
```

Comment:
So this is trying to support multiple .env to override env variables? I haven't seen many examples where we need multiple .env files? Is it because each extension might need to override env variables?

Reply:
Yes, this is to support extensions. But also to support multiple instances with different configuration.

.env - common stuff like API keys, maybe also common objective
.env.instance1 - initial task 1
.env.instance2 - initial task 2, different Pinecone table
.env.instance3 - different execution agent

Now you can easily run and compare side-by-side how the initial task or execution agent change affects how it works.